webgeocalc/calculation.py

```diff
@@ -1709,3 +1710,175 @@ def direction_vector_dec(self, val):
+        '''The amount by which to expand or contract each interval at the endpoints.
+        Each endpoint will be moved by this amount.
+
+        Parameters
+        ----------
+        interval_adjustment_amount: float
+
+        '''
+        self.__intervalAdjustmentAmount = val
+
+    @SetterProperty
+    def interval_adjustment_units(self, val):
+        '''The unit of the interval adjustment amount
+
+        Parameters
+        ----------
+        interval_adjustment_units: str
+            One of the following:
+
+            - SECONDS
+            - MINUTES
+            - HOURS
+            - DAYS
+
+        Raises
+        -------
+        CalculationInvalidAttr
+            If the value provided is invalid.
+        '''
+        if val in INTERVAL_ADJUSTMENT_UNITS:
+            self.__intervalAdjustmentUnits = val
+        else:
+            raise CalculationInvalidAttr('interval_adjustment_units', val, INTERVAL_ADJUSTMENT_UNITS)
+
+    @SetterProperty
+    def interval_filtering(self, val):
+        '''Specifies whether to omit interval smaller than a minimum threshold size. This threshold is applied after
+        expansion or contraction of the intervals.
+
+        Parameters
+        ----------
+        interval_filtering: str
+            One of the following:
+
+            - NO_FILTERING
+            - OMIT_INTERVALS_SMALLER_THAN_A_THRESHOLD
+
+        Raises
+        -------
+        CalculationInvalidAttr
+            If the value provided is invalid.
+        '''
+        if val in INTERVAL_FILTERING:
+            self.__intervalFiltering = val
+        else:
+            raise CalculationInvalidAttr('interval_filtering', val, INTERVAL_FILTERING)
+
+    @SetterProperty
+    def interval_filtering_threshold(self, val):
+        '''Interval duration filtering threshold value
+
+        Parameters
+        ----------
+        interval_filtering_threshold: float
+            Interval duration filtering threshold value
+        '''
+        self.__intervalFilteringThreshold = val
+
+    @SetterProperty
+    def interval_filtering_threshold_units(self, val):
+        '''Units of the interval duration filtering threshold value
+
+        Parameters
+        ----------
+        interval_filtering_threshold_units: str
+            One of the following:
+
+            - SECONDS
+            - MINUTES
+            - HOURS
+            - DAYS
+
+        Raises
+        -------
+        CalculationInvalidAttr
+            If the value provided is invalid.
+        '''
+        if val in INTERVAL_FILTERING_THRESHOLD_UNITS:
+            self.__intervalFilteringThresholdUnits = val
+        else:
+            raise CalculationInvalidAttr('interval_filtering_threshold_units', val, INTERVAL_FILTERING_THRESHOLD_UNITS)
+
+    @SetterProperty
+    def condition(self, val):
+        '''Coordinate condition for the Coordinate Search
+
+        Parameters
+        ----------
+        condition: dict
+        '''
+        self.__condition = val
```

Comment:
The input `condition` needs to be valided

Reply:
This has been reworked.